libflux/src/core/ast/tests.rs

```diff
@@ -395,6 +395,21 @@ fn test_json_expression_statement() {
     let deserialized: Statement = serde_json::from_str(serialized.as_str()).unwrap();
     assert_eq!(deserialized, n)
 }
+
+#[test]
+fn test_json_namedtype() {
+    let n = MonoType::Basic(NamedType {
+        base: BaseNode::default(),
+        name: Identifier {
+            base: BaseNode::default(),
+            name: "int".to_string(),
+        },
+    });
+    let serialized = serde_json::to_string(&n).unwrap();
+    assert_eq!(serialized, r#"{"Basic":{"name":{"name":"int"}}}"#);
```

Comment:
I think this should look like:

    {"type":"NamedType","name":{"type":"Identifier","name":"int"}}

In particular, every json element should have a `type` field. In order to do that you'll need to add `#[serde(tag = "type")]` to the MonoType enum. So the MonoType enum should look like this:
```Rust
#[serde(tag = "type")]
pub enum MonoType {
    #[serde(rename = "NamedType")]
    Basic(NamedType),
    ....
```

Reply:
Do I replace #[derive(Debug, PartialEq, Clone, Serialize, Deserialize)] with #[serde(tag = "type")] or keep both?

Reply:
Keep both. You can add `#[serde(tag = "type")]` right below the first one.